CoreFoundation/Base.subproj/CFInternal.h

```diff
@@ -279,6 +277,24 @@ static inline void __CFRuntimeSetFlag(CFTypeRef cf, uint8_t n, Boolean flag) {
 CF_PRIVATE Boolean __CFRuntimeIsConstant(CFTypeRef cf);
 CF_PRIVATE void __CFRuntimeSetRC(CFTypeRef cf, uint32_t rc);
 
+#if DEPLOYMENT_RUNTIME_SWIFT
+#define _CFRUNTIME_BASE_INIT_SWIFT_RETAIN_COUNT ._swift_rc = _CF_CONSTANT_OBJECT_STRONG_RC
+#else
+#define _CFRUNTIME_BASE_INIT_SWIFT_RETAIN_COUNT
+#endif
+
+// A note on these macros.
+// On systems where we have ObjC support (DEPLOYMENT_RUNTIME_OBJC), STATIC_CLASS_REF(…) can statically produce a reference to the ObjC class symbol that ties into this particular type.
+// When compiling for Swift Foundation, STATIC_CLASS_REF returns a Swift class. There's a mapping of ObjC name classes to Swift symbols in the header that defines it that should be kept up to date if more constant objects are defined.
+// On all other platforms, it returns NULL, which is okay; we only need the type ID if CF is to be used by itself.
+#if __LP64__
```

Comment:
This set of macros is a little weird for a couple reasons:

It is using a ULL in the 32-bit portion.

It assumes that Darwin and non-Darwin both use effectively “unsigned” as the cfinfoa type. Probably not fatal, but confusing to read.

Also curious why this was added here rather than where the other initialization macros are, and why do we need another set?

Out of curiosity, does anyone know if _CFInfo is only used on Linux (non-ObjC) targets? Seems like it is?

Reply:
`_cfinfoa` is used on all targets. Its bits correspond to the ObjC object flag word semantics (please, runtime people, correct me if I'm wrong?), and it is used to store the refcount in ObjC and standalone (non-Swift) builds.

Reply:
Okay, so these macros add more parameters that `INIT_CFRUNTIME_BASE` in `CFRuntime.h` doesn't. That's fine. It's just a little hard to compare the two sets of macros that do incredibly similar things because they are in different files. But since this is for internal, I guess that's the point.

My comment about _cfinfoa is more that on 32-bit, you've got a `ULL` (64-bit) instead of a `UL` (32-bit) const. 

As for _CFInfo, I bring it up because it's padded to 64-bits, at least on Linux, which makes macros like this... interesting.